streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -107,7 +108,13 @@
         this.builder = builder;
         this.adminClient = adminClient;
         this.stateDirectory = stateDirectory;
-        this.eosEnabled = eosEnabled;
+        if (StreamThread.eosAlphaEnabled(config)) {
```

Comment:
We have duplicated logic here and in TaskCreator to define the processingMode, could we just let the thread decide and then pass the decision along in callee constructors?

Reply:
Maybe -- that would lead to more coupling or we need to pass more parameters around? I like to reduce the number of parameters we need to forward and we have `config` already at hand for many other reasons. Strictly, we should never forward `config`, but extract the config we need and only pass those along -- but it blows up the code.

Thought?

Reply:
For TaskCreator and TaskManager here's my thoughts: ideally the taskCreators could be constructed inside task-manager, but for unit testings we want to mock the creators sometimes so we create them outside of task-manager, inside stream-thread. So for this special case I think passing the ProcessMode into these two classes are okay, since for most other cases the hierarchy is natural and we would only see the construction of such parameters in a single place.